implementation/tracing-propagation/src/main/java/io/smallrye/faulttolerance/tracing/TracingContextProvider.java

```diff
@@ -12,43 +12,44 @@
 import io.opentracing.Tracer;
 import io.opentracing.util.GlobalTracer;
 
-/**
- * @author Michal Szynkiewicz, michal.l.szynkiewicz@gmail.com
- */
 public class TracingContextProvider implements ThreadContextProvider {
-
     private static final ThreadContextController DO_NOTHING = () -> {
     };
 
     @Override
     public ThreadContextSnapshot currentContext(Map<String, String> props) {
         Tracer tracer = GlobalTracer.get();
         ScopeManager scopeManager = tracer.scopeManager();
-        Scope activeScope = scopeManager.active();
+        Span span = scopeManager.activeSpan();
 
-        if (activeScope != null) {
-            Span span = activeScope.span();
+        if (span != null) {
             return () -> {
-                Scope propagated = scopeManager.activate(span, false);
+                Scope propagated = scopeManager.activate(span);
                 return propagated::close;
             };
         }
```

Comment:
I'm looking at this again and I found this a bit weird. If the original thread had an active span, but not an active scope, the new thread will have the same active span, but it will _also_ have an active scope.

I have read through https://github.com/opentracing/opentracing-java/issues/267 and https://github.com/opentracing/opentracing-java/issues/291 and it seems to me that OpenTracing just actively fights against our progamming model :-)

But I guess there's nothing better that we could do.